### DISCUSSION TITLE
Code works fine on android but  not on iOS

### DISCUSSION THREAD

**Discussion by lotalogic** [innoveit/react-native-ble-manager#1184] [Q&A]
Hi,
I am using a gatt server running on my raspberry pi and I have developed a react native app based on the react-native-ble-manager/example.
Everything works fine on android but when I test on iOs a popup opens on my raspberry pi about a pairing code and same message on iOs and even if I click Ok on both parts it fails.
How can I get rid off this pairing on iOs ?
When running the server on my desktop (instead or the rpi) I don't have the popup about pairing but doesn't work either.

Here is the gatt server I use:
[https://github.com/lotalogic/debug-ble-server-ios](https://github.com/lotalogic/debug-ble-server-ios)
You install rust
curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs/ | sh
Then you exit terminal and start a new one.
cd debug-ble-server-ios
cargo build
./target/debug/gatt_remotectl_server and then try to connect with an app using react-native-ble-manager.

here is the important UUID of the server:

const knownServiceUUID = '00000000-0000-0000-0000-0000feedc0de';
const knownCharacteristicUUID = '00000000-0000-0000-000f-00dc0de00001';

Would be curious to see if you manage to do some write on iOs.

When I try I have the following logs:

Connecting to peripheral with UUID: 187d2848-1a38-abfd-7fc0-0412abb34744
Peripheral Connected: 187d2848-1a38-abfd-7fc0-0412abb34744
[connectPeripheral][187d2848-1a38-abfd-7fc0-0412abb34744] connected.
[handleConnectPeripheral][187d2848-1a38-abfd-7fc0-0412abb34744] connected.
write
Could not find service with UUID 00000000-0000-0000-0000-0000feedc0de on peripheral with UUID 187d2848-1a38-abfd-7fc0-0412abb34744.

**Comment by lotalogic** [comment]
Maybe I should try with python bluez first in case the problem is inside bluez rust wrapper ...

UPDATE:
I have updated the repository I use to test and this time I am testing with  bluez in python and handling the Agent/authorization thing, so once again it works fine on android and not on iOS......
[android-ok.txt](https://github.com/innoveit/react-native-ble-manager/files/14619873/android-ok.txt)
[ios-ko.txt](https://github.com/innoveit/react-native-ble-manager/files/14619874/ios-ko.txt)
And when attaching Xcode debugger I only see the same error message: Could not find service with UUID ....
If someone wants to test ,please clone my debug repo:
cd bluez
python3 app.py

and try to write a value...

**Comment by marcosinigaglia** [comment]
The pairing is starting because you have some encrypted characteristic in the gatt server.

**Comment by lotalogic** [comment]
Actually none of the didDiscover functions are called only connect and then a few seconds later a disconnect and then I have to go to bluetooth settings to press forget this device if I want to connect again.
So to me there is an issue with iOS code.

**Reply by lotalogic** [reply]
I am progressing, one of the first problem is the fact I was calling await BluetoothService.connect(peripheral.id) instead of await BluetoothService.connect(peripheral.id, true) !!!! and on iOs if you don"t pass true it doesn't discover services...
